### PR TITLE
PSQLADM-373 Update the docs based on added features

### PR DESCRIPTION
**Pull request by patrickbirch**
D  source/_res/code-blocks/select-from-mysql-servers.txt
D  source/_res/tables/proxysql-pxc.txt
M  source/configuring.rst
M  source/index.rst
A  source/install-v2.rst
A  source/proxysql-changes.rst
D  source/proxysql-v2.rst
A  source/psql-functions.rst
A  source/v2-config.rst
A  source/v2-scheduler.rst

### CODE REVIEW

File: source/configuring.rst [hunk 1]
```diff
@@ -80,38 +113,38 @@ To see the |proxysql| databases and tables use the following commands:
...
   * MEMORY (your current working place)
   * RUNTIME (the production settings)
-  * DISK (durable configuration, saved inside an SQLITE database)
+  * DISK (durable configuration saved in a `SQLite <https://sqlite.org/index.html>`__ database)
 
-  When you change a parameter, you change it in MEMORY area.  That is done by
-  design to allow you to test the changes before pushing to production
-  (RUNTIME), or save them to disk.
+  When you change a parameter, you change it in the MEMORY area.  This method
+  allows you to test the changes before pushing to the change to production
```
Comment (alina-derkach-oaza) [inline source/configuring.rst:128]: before pushing to the change -> before pushing the change

File: source/configuring.rst [hunk 2]
```diff
@@ -80,38 +113,38 @@ To see the |proxysql| databases and tables use the following commands:
...
 
   * MEMORY (your current working place)
   * RUNTIME (the production settings)
-  * DISK (durable configuration, saved inside an SQLITE database)
+  * DISK (durable configuration saved in a `SQLite <https://sqlite.org/index.html>`__ database)
 
-  When you change a parameter, you change it in MEMORY area.  That is done by
-  design to allow you to test the changes before pushing to production
-  (RUNTIME), or save them to disk.
+  When you change a parameter, you change it in the MEMORY area.  This method
```
Comment (alina-derkach-oaza) [inline source/configuring.rst:127]: area.  This - remove extra space

File: source/configuring.rst [hunk 3]
```diff
@@ -40,9 +56,26 @@ specified in the `ProxySQL global variables`_.
...
-.. include:: _res/code-blocks/mysql-hpu.txt
+.. code-block:: bash
+
+   root@proxysql:~# mysql -u admin -padmin -h 127.0.0.1 -P 6032
+
+   Welcome to the MySQL monitor.  Commands end with ; or \g.
+   Your MySQL connection id is 2
+   Server version: 5.1.30 (ProxySQL Admin Module)
+
+   Copyright (c) 2009-2016 Percona LLC and/or its affiliates
```
Comment (alina-derkach-oaza) [inline source/configuring.rst:67]: 2009-2016 - should it be 2009-2022?

File: source/configuring.rst [hunk 4]
```diff
@@ -40,9 +56,26 @@ specified in the `ProxySQL global variables`_.
...
+   Copyright (c) 2000, 2016, Oracle and/or its affiliates. All rights reserved.
```
Comment (alina-derkach-oaza) [inline source/configuring.rst:68]: 2009, 2016 - should it be 2009-2022?

File: source/configuring.rst [hunk 5]
```diff
@@ -80,38 +113,38 @@ To see the |proxysql| databases and tables use the following commands:
...
 .. note::
 
-   |proxysql| uses the concept of *hostgroups* to group cluster nodes. This
+   *ProxySQL* uses the concept of *hostgroups* to group cluster nodes. This
    approach enables balancing the load in a cluster by routing different types
    of traffic to different groups.
 
    There are many ways you can configure hostgroups (for
-   example master and slaves, read and write load, etc.)  and every node can
+   example, master and slaves, read and write load, etc.)  and a node can
```
Comment (alina-derkach-oaza) [inline source/configuring.rst:144]: etc.)  and -> remove extra space

File: source/configuring.rst [hunk 6]
```diff
@@ -466,10 +507,19 @@ Assisted Maintenance Mode
 Usually, to take a node down for maintenance, you need to identify that node,
 update its status in ProxySQL to ``OFFLINE_SOFT``, wait for ProxySQL to divert
 traffic from this node, and then initiate the shutdown or perform maintenance
-tasks.  |pxc| includes a special *maintenance mode* for nodes that enables you
-to take a node down without adjusting |proxysql| manually.  This mode is
-controlled via `pxc_maint_mode variable`_, which is monitored by
-|proxysql| and can be set to one of the following values:
+tasks.  *Percona XtraDB Cluster* includes a special *maintenance mode* for nodes that enables you
+to take a node down without adjusting *ProxySQL* manually.  This mode is
```
Comment (alina-derkach-oaza) [inline source/configuring.rst:511]: manually.  This -> remove extra space

File: source/proxysql-changes.rst [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  server nodes as found. (Note that nodes are added but not removed).  The
+  ``--writer-hg`` option may be used to specify which Galera hostgroup to
+  update. The ``--remove-all-servers`` option instructs to remove all servers
+  from the mysql_servers table before updating the cluster.
+- Hostgroups can be specified on the command-line: ``--writer-hg``,
+  ``--reader-hg``, ``--backup-writer-hg``, and ``--offline-hg``.  Previously,
+  these host groups were only read from the configuration file.
+- The ``--enable`` and ``--update-cluster`` options used simultaneously have
+  special meaning. If the cluster has not been enabled, then ``--enable`` is
+  run.  If the cluster has already been enabled, then ``--update-cluster`` is
```
Comment (alina-derkach-oaza) [inline source/proxysql-changes.rst:25]: run.  If -> remove extra space

File: source/proxysql-changes.rst [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  run.
+- New command ``--is-enabled`` to see if a cluster has been enabled. This
+  command checks for the existence of a row in the mysql_Galera_hostgroups
+  table.  The ``--writer-hg`` option may be used to specify the writer hostgroup
```
Comment (alina-derkach-oaza) [inline source/proxysql-changes.rst:29]: table.  The -> remove extra space

File: source/proxysql-changes.rst [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+Changed Features
+--------------------------------------------------------------------------------
+
+- Setting ``--node-check-interval`` changes the ProxySQL global variable
+  ``mysql-monitor_galera_healthcheck_interval``. Note that this is a global
+  variable, not a per-cluster variable.
+- The option ``--write-node`` takes only a single address as a parameter. In the
+  singlewrite mode we only set the weight if ``--write-node`` specifies
+  *address:port*.  A priority list of addresses is no longer accepted.
```
Comment (alina-derkach-oaza) [inline source/proxysql-changes.rst:52]: *address:port*.  A priority -> remove extra space

File: source/proxysql-changes.rst [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  longer run in the scheduler, automatic cluster membership updates are not
+  supported.
+- Checking the pxc_maint_mode variable is no longer supported
+- Using desynced nodes if no other nodes are available is no longer supported.
+- The server status is no longer maintained in the mysql_servers table.
+
+Limitations
+--------------------------------------------------------------------------------
+
+- With ``--writers-as-readers=backup`` read-only nodes are not allowed. This a
```
Comment (alina-derkach-oaza) [inline source/proxysql-changes.rst:87]: This a -> this is a

File: source/psql-functions.rst [hunk 11]
```diff
@@ -0,0 +1,501 @@
...
+
+A specific Galera cluster can be disabled by using the --writer-hg option with
+``--disable``.
+
+.. _pxc.proxysql.v2.admin-tool.adduser:
+
+.. option:: --adduser
+
+This option will aid with adding the Cluster application user to the ProxySQL
+database for you
```
Comment (alina-derkach-oaza) [inline source/psql-functions.rst:119]: for you -> for you.

File: source/configuring.rst (hunk 1 above)
Comment (patrickbirch) [reply source/configuring.rst:128]: Fixed.

File: source/configuring.rst (hunk 2 above)
Comment (patrickbirch) [reply source/configuring.rst:127]: Done.

File: source/configuring.rst (hunk 3 above)
Comment (patrickbirch) [reply source/configuring.rst:67]: It's in the code.

File: source/configuring.rst (hunk 5 above)
Comment (patrickbirch) [reply source/configuring.rst:144]: Fixed.

File: source/configuring.rst (hunk 6 above)
Comment (patrickbirch) [reply source/configuring.rst:511]: Fixed.

File: source/proxysql-changes.rst (hunk 7 above)
Comment (patrickbirch) [reply source/proxysql-changes.rst:25]: Done.

File: source/proxysql-changes.rst (hunk 8 above)
Comment (patrickbirch) [reply source/proxysql-changes.rst:29]: Done.

File: source/proxysql-changes.rst (hunk 9 above)
Comment (patrickbirch) [reply source/proxysql-changes.rst:52]: Done.

File: source/proxysql-changes.rst (hunk 10 above)
Comment (patrickbirch) [reply source/proxysql-changes.rst:87]: Added.

File: source/psql-functions.rst (hunk 11 above)
Comment (patrickbirch) [reply source/psql-functions.rst:117]: Added period

File: source/v2-scheduler.rst [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+
+.. rubric:: The percona-scheduler-admin usage info
+
+.. sourcecode:: bash
+
+   percona-scheduler-admin [ options ]
+   
+.. seealso:: :ref:`psa-options` for examples of common operations.
+
+You must include at least one options. The options are either independent, do not require other options, or are dependent on other options.
```
Comment (alina-derkach-oaza) [inline source/v2-scheduler.rst:46]: options -> option

Comment (alina-derkach-oaza) [inline source/v2-scheduler.rst:46]: The options are either independent, do not require other options, -> I would add either "and" here or ()

"The options are either independent and do not require other options,"
or 
"The options are either independent (do not require other options),"

File: source/v2-scheduler.rst [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+   * - ``--status``
+     - Returns a status report on the current configuration.
+   * - ``--server=<IPADDRESS>:<PORT>``   
+     - Specifies the IP address and port for a single server.
+   * - ``--trace``                            
+     - Enables shell-level tracing for this shell script.
+   * - ``--use-stdin-for-credentials``
+     - Uses stdin to send credential to the MySQL client instead of process substitution. The default disables the option and process substitution is used.
+   * - ``--version``, ``-v``                      
+     - Prints the version info
```
Comment (alina-derkach-oaza) [inline source/v2-scheduler.rst:84]: add a full stop

File: source/v2-scheduler.rst [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+
+
+The following options require other options or a specific mode. If you run these options by themselves or with incorrect options causes an error.
```
Comment (alina-derkach-oaza) [inline source/v2-scheduler.rst:87]: If you run these options by themselves or with incorrect options causes an error. -> Running these options by themselves or with incorrect options causes an error.

File: source/v2-scheduler.rst [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+     - Requires mode
+     - Description
+   * - ``--add-query-rule`` 
+     - ``--syncuser``, ``--sync-multi-cluster-users``
+     - ``singlewrite``
+     - Creates query rules for the synced mysql user.
+   * - ``--auto-assign-weights``
+     - ``--update-cluster``
+     -  ``singlewrite``
+     - Auto-assigns the weights
```
Comment (alina-derkach-oaza) [inline source/v2-scheduler.rst:103]: add a full stop

File: source/v2-scheduler.rst [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+
+==========================================================
+ProxySQL and Percona Scheduler Admin
+==========================================================
+
+Implemented in ProxySQL 2.3.2-1, the Percona Scheduler Admin (percona-scheduler-admin) utility configures **Percona XtraDB Cluster** nodes into *ProxySQL*. 
+
+The Percona Scheduler Admin works with ProxySQL 2.3.0.  
+
+The utility has an external scheduler which is capable of the following operations:
```
Comment (alina-derkach-oaza) [inline source/v2-scheduler.rst:11]: which -> that

File: source/v2-scheduler.rst [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+* Parallel query execution on nodes results in faster failover or fallback
+* Supports fallback
+* Supports promoting a reader to primary with the following conditions are met:
+    * wsrep_on=OFF
+    * wsrep_sst_donor_rejects_queries is set when, for example, connections are rejected due to an ongoing state transfer
+* Supports a retry of operations in case of network glitches
+
+This tool can be used to automatically perform a failover due to node failures, service degradation, or maintenance. 
+
+Please log ProxySQL Admin bug reports here: https://jira.percona.com/projects/PSQLADM.
```
Comment (alina-derkach-oaza) [inline source/v2-scheduler.rst:23]: I would make "here" a link -> Please log ProxySQL Admin bug reports `here <https://jira.percona.com/projects/PSQLADM>`__.

File: source/v2-scheduler.rst (hunk 12 above)
Comment (patrickbirch) [reply source/v2-scheduler.rst:46]: Rewritten

Comment (patrickbirch) [reply source/v2-scheduler.rst:46]: Rewritten.

File: source/v2-scheduler.rst [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+
+.. list-table:: Independent options 
+   :header-rows: 1
+   
+   * - Option
+     - Description
+
+
+   * - `--adduser`
+     - Adds the Percona XtraDB Cluster application user to the ProxySQL database
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/v2-scheduler.rst:59]: Done.

File: source/v2-scheduler.rst (hunk 13 above)
Comment (patrickbirch) [reply source/v2-scheduler.rst:84]: Done.

File: source/v2-scheduler.rst (hunk 14 above)
Comment (patrickbirch) [reply source/v2-scheduler.rst:87]: Rewritten

File: source/v2-scheduler.rst (hunk 15 above)
Comment (patrickbirch) [reply source/v2-scheduler.rst:103]: Added.

File: source/v2-scheduler.rst (hunk 16 above)
Comment (patrickbirch) [reply source/v2-scheduler.rst:11]: that

File: source/v2-scheduler.rst (hunk 17 above)
Comment (patrickbirch) [reply source/v2-scheduler.rst:23]: Rewritten.

**Review by alina-derkach-oaza — APPROVED** [review]

File: source/configuring.rst [hunk 19]
```diff
@@ -337,18 +378,18 @@ You can install ``sysbench`` from Percona software repositories:
...
 
 * For Red Hat Enterprise Linux or CentOS
 
   .. code-block:: bash
 
-     [root@proxysql ~]# yum install sysbench
+     [root@proxysql ~]#> yum install sysbench
 
 .. note:: ``sysbench`` requires ProxySQL client user credentials
    that you creted in :ref:`proxysql-client-user`.
```
Comment (venkatesh-prasad-v) [inline source/configuring.rst:390]: s/creted/created/

File: source/index.rst [hunk 20]
```diff
@@ -1,44 +1,100 @@
 .. _index.proxysql-admin-interface:
 
-.. ProxySQL Admin Documentation documentation master file, created by
-   sphinx-quickstart on Fri Apr 17 21:29:13 2020.
-   You can adapt this file completely to your liking, but it should at least
-   contain the root `toctree` directive.
 
-|proxysql| and |proxysql-admin| documentation
+*ProxySQL*, **proxysql-admin**, and percona-scheduler documentation
```
Comment (venkatesh-prasad-v) [inline source/index.rst:4]: https://psqladm-373.onrender.com/index.html#:~:text=Edit%20this%20page-,ProxySQL%2C%20proxysql%2Dadmin%2C%20and%20percona%2Dscheduler%20documentation,%C2%B6,-ProxySQL%20is%20a

This makes all three product names to look differently. Is this expected?

File: source/known-limitations.rst [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+
+Do not combine the following options
+---------------------------------------------------
+
+The following options are mutually exclusive. An attempt to run them together
+results in error.
+
+*  :ref:`psa-update-write-weight` and :ref:`psa-auto-assign-weights`
+
+*  ``--write-node`` and :ref:`psa-auto-assign-weights`
```
Comment (venkatesh-prasad-v) [inline source/known-limitations.rst:18]: `--write-node` does not have a hyperlink.

File: source/psa-build.rst [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+configuration file for the ``pxc-scheduler_handler`` binary.
+
+Running ``pxc-scheduler-handler``  without ``percona_scheduler_admin`` requires
+manual intervention.
+
+We do not recommend running multiple instances of the same binary. If you
+attempt this task when an instance of
+``pxc-scheduler-handler`` is running the binary runs, but the following occurs:
+
+* Provide information manually
```
Comment (venkatesh-prasad-v) [inline source/psa-build.rst:40]: This is wrong. We can remove this line.

File: source/psa-config.rst [hunk 23]
```diff
@@ -0,0 +1,178 @@
...
+  # ProxySQL will use SSL to connect to the backend servers
+  #
+  useSSL=0
+
+  # --------------------------------
+  # Max number of connections from ProxySQL to the backend servers.
+  #
+  maxConnections=1000
+
+  # Sets the proxysql's mysql-monitor_galera_healthcheck_interval variable
```
Comment (venkatesh-prasad-v) [inline source/psa-config.rst:175]: This is wrong. This value given here is passed on proxysql's `interval_ms` field in the `scheduler` table.
In simple words, this value is used by the percona-scheduler-admin script and causes the pxc_scheduler_handler to be run every `nodeCheckInterval` milliseconds.

**Review by venkatesh-prasad-v — CHANGES_REQUESTED** [review]

File: source/psa-config.rst (hunk 23 above)
Comment (patrickbirch) [reply source/psa-config.rst:175]: This information is taken from the example configuration file in https://github.com/percona/proxysql-admin-tool
Fix that please.

File: source/psa-build.rst [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+
+Create an ``admin`` user account. Use this account for communication through
+proxysql and pxc_scheduler_handler.
+
+.. sourcecode::
+
+
+   mysql> CREATE USER 'admin'@'192.%' IDENTIFIED WITH 'mysql_native_password'
+   by
+   'admin';
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/psa-build.rst]: Doesn't fit one line. Redid it to two lines.

File: source/psa-build.rst (hunk 22 above)
Comment (patrickbirch) [reply source/psa-build.rst:40]: Remove how much? Just the Provide information manually line or everything from line 37 to 40?

File: source/psa-build.rst [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+attempt this task when an instance of
+``pxc-scheduler-handler`` is running the binary runs, but the following occurs:
+
+* Provide information manually
+* Consumes network and system resources
+* Returns the same results since the multiple versions use the same
+configuration file
+
+
+The ``pxc_scheduler_hander`` binary is located in the base directory.
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/psa-build.rst:46]: Fixed.

File: source/configuring.rst (hunk 19 above)
Comment (patrickbirch) [reply source/configuring.rst:390]: v1 - I'll create a ticket to fix this.

File: source/configuring.rst [hunk 26]
```diff
@@ -270,7 +311,7 @@ The following example shows how you can load the script for default
 
 .. code-block:: text
 
-   INSERT INTO scheduler (active,interval_ms,filename,arg1,comment)
+   mysql> INSERT INTO scheduler (active,interval_ms,filename,arg1,comment)
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/configuring.rst:314]: v1 - create a ticket to update this.

File: source/known-limitations.rst (hunk 21 above)
Comment (patrickbirch) [reply source/known-limitations.rst:18]: --write-node is not listed separately on the Options page, therefore, there is no anchor to link to.

File: source/index.rst (hunk 20 above)
Comment (patrickbirch) [reply source/index.rst:4]: Fixed.

File: source/psa-build.rst [hunk 27]
```diff
@@ -0,0 +1,59 @@
+.. _psa-build:
+
+=============================================
+Build the Percona Scheduler admin tool
+=============================================
+
+In the command line, run the following statement initialize all
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/psa-build.rst:7]: Fixed.

File: source/psa-build.rst [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+manual intervention.
+
+We do not recommend running multiple instances of the same binary. If you
+attempt this task when an instance of
+``pxc-scheduler-handler`` is running the binary runs, but the following occurs:
+
+* Provide information manually
+* Consumes network and system resources
+* Returns the same results since the multiple versions use the same
+configuration file
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/psa-build.rst:43]: 80 character limit. :)

File: source/configuring.rst [hunk 29]
```diff
@@ -1,22 +1,38 @@
...
+     - pxc2
+     - 192.168.70.62
+   * - Node 3
+     - pxc3
+     - 192.168.70.63
+   * - Node 4
+     - proxysql
+     - 192.168.70.64 
+
+*ProxySQL* can be configured either using *etc/proxysql-admin.cnf* or using the
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/configuring.rst:29]: I'll make a ticket to fix this later. Not related to psa material.

File: source/configuring.rst [hunk 30]
```diff
@@ -466,10 +507,19 @@ Assisted Maintenance Mode
 Usually, to take a node down for maintenance, you need to identify that node,
 update its status in ProxySQL to ``OFFLINE_SOFT``, wait for ProxySQL to divert
 traffic from this node, and then initiate the shutdown or perform maintenance
-tasks.  |pxc| includes a special *maintenance mode* for nodes that enables you
-to take a node down without adjusting |proxysql| manually.  This mode is
-controlled via `pxc_maint_mode variable`_, which is monitored by
-|proxysql| and can be set to one of the following values:
+tasks.  *Percona XtraDB Cluster* includes a special *maintenance mode* for nodes that enables you
+to take a node down without adjusting *ProxySQL* manually. This mode is
+controlled via `pxc_maint_mode_` variable, which is monitored by
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/configuring.rst:512]: Not related to PSA material. Will fix at a later time.

File: source/psa-build.rst (hunk 22 above)
Comment (venkatesh-prasad-v) [reply source/psa-build.rst]: Remove the only line that says `* Provide information manually`

File: source/psa-config.rst (hunk 23 above)
Comment (venkatesh-prasad-v) [reply source/psa-config.rst:175]: We have fixed it in the [release branch](https://github.com/percona/proxysql-admin-tool/tree/v2.3.2-1-dev). So, it is not visible in the default(v2) branch.

Comment (patrickbirch) [reply source/psa-config.rst]: Copied latest version

**Review by venkatesh-prasad-v — CHANGES_REQUESTED** [review]

File: source/psa-options.rst [hunk 31]
```diff
@@ -0,0 +1,585 @@
...
+    Removing cluster application users from the ProxySQL database.
+    Removing cluster nodes from the ProxySQL database.
+    Removing query rules from the ProxySQL database if any.
+
+.. _psa-enable:
+
+--enable / -e
+^^^^^^^^^^^^^^^^^
+
+This option creates the entry for the Galera hostgroups and adds the Percona XtraDB Cluster nodes into ProxySQL.
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/psa-options.rst:180]: Updated.

File: source/psa-options.rst [hunk 32]
```diff
@@ -0,0 +1,585 @@
...
+    | monitor  |
+    +----------+
+    1 row in set (0.00 sec)
+
+
+    # Add a new user on a PXC node
+      PXC node> mysql> SELECT user FROM mysql.user WHERE user LIKE 'test%';
+    Empty set (0.00 sec)
+
+    mysql> CREATE USER 'test_user'@'localhost' IDENTIFIED WITH 'mysql_native_password' by 'passw0Rd';
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/psa-options.rst:370]: This change does not fit with the documentation standards, but it fine for your README.

File: source/psa-options.rst [hunk 33]
```diff
@@ -0,0 +1,585 @@
...
+    Query OK, 0 rows affected (0.04 sec)
+
+
+    # Run percona-scheduler-admin with --syncusers
+    ./percona-scheduler-admin --config-file=config.toml --syncusers
+
+    Syncing user accounts from PXC(127.0.0.1:4130) to ProxySQL
```
[the remark this answers is not in the document]
Comment (patrickbirch) [reply source/psa-options.rst:377]: OK.

**Pull request merged**